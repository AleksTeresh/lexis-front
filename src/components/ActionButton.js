import React from 'react'
import Button from 'material-ui/Button'
import styled from 'styled-components'

const AddButton = styled(Button)`
  position: fixed !important;
  bottom: 50px;
  right: 50px;
`
type ActionButtonProps = {
  children: any,
  color?: string,
  ariaLabel?: string
}
export default function ActionButton ({
  children,
  color = 'primary',
  ariaLabel = 'add'
}: ActionButtonProps) {
  return (
    <AddButton fab color={color} aria-label={ariaLabel}>
      {children}
    </AddButton>
  )
}
