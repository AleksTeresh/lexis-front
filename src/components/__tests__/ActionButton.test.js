/* eslint-env jest */

import React from 'react'
import ActionButton from '../ActionButton'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import { configure } from 'enzyme'
import Adapter from 'enzyme-adapter-react-15'

configure({ adapter: new Adapter() })

describe('ActionButton Component', () => {
  it('renders a snapshot', () => {
    const wrapper = shallow(<ActionButton />)

    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
