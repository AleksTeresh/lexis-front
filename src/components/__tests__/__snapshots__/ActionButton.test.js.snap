// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActionButton Component renders a snapshot 1`] = `
<withStyles(Button)
  aria-label="add"
  color="primary"
  fab={true}
  onClick={[Function]}
  style={
    Object {
      "bottom": "50px",
      "position": "fixed",
      "right": "50px",
    }
  }
/>
`;
